src/lib/holocene/select/select.svelte

```diff
@@ -113,14 +115,16 @@
     <MenuButton
       class={twMerge('w-full', !valid ? 'border-danger' : undefined)}
       hasIndicator={!disabled}
-      {disabled}
+      disabled={disabled || loading}
       controls="{id}-select"
       {variant}
       data-testid={`${$$restProps['data-testid'] ?? ''}-button`}
     >
       <slot name="leading" slot="leading">
         {#if leadingIcon}
           <Icon name={leadingIcon} />
+        {:else if loading}
+          <Icon name="spinner" class="animate-spin" />
         {/if}
```

Comment:
Should we try to keep the input the same width even when it's loading so it doesn't have that jump when switching between loading/not loading?

Reply:
Also wondering if we could just pass `loading` to the `MenuButton` since the underlining `Button` component takes it as a prop.

Reply:
I'm not sure how we would do that within the component itself, other than specifying a fixed on the MenuContainer (which we probably shouldn't do). Currently you can add a fixed with to the parent element and it won't grow.


https://github.com/user-attachments/assets/918fb235-391a-48a9-bef1-e299f602dced

Reply:
Although I don't like how the text still jumps. I thought about replacing the chevron icon with the spinner, but Button has the spinner leading the text, so I wanted to keep it consistent

Reply:
How about this?

https://github.com/user-attachments/assets/b6ff66a4-ee37-4a17-bccd-165e1779ab74

Reply:
I like that

Reply:
@laurakwhit 

> Also wondering if we could just pass loading to the MenuButton since the underlying Button component takes it as a prop.

Good ideal, although currently `<Button />` doesn't disable itself when `loading` is true, and I want that behavior for `<Select />`. I thought `<Button />` used to do that, but maybe not.